packages/builder/src/stores/builder/components.ts

```diff
@@ -33,17 +33,28 @@ import { Utils } from "@budibase/frontend-core"
 import { Component, FieldType, Screen, Table } from "@budibase/types"
 import { utils } from "@budibase/shared-core"
 
-interface ComponentDefinition {
+export interface ComponentState {
+  components: Record<string, ComponentDefinition>
+  customComponents: string[]
+  selectedComponentId?: string | null
```

Comment:
We should probably use either an optional property or type it as `string | null` shouldn't we? Rather than both. I've been opting to go for optional properties - the only change being we'll need to ensure we always clear these by setting them to `undefined` rather than `null`.